### DISCUSSION TITLE
Choosing a hubverse "dashboard" component to tackle for the upcoming season

### DISCUSSION THREAD

**Discussion by bsweger** [hubverse-org/.github#23] [Polls]
**TL;DR**

We would like to have a preliminary/beta hub dashboard ready for the upcoming season (_i.e._, autumn, here in the Northern Hemisphere), with the goal of getting early feedback from hubs willing to test the feature and provide feedback. Thus, we need to decide what type of dashboard/component to tackle first.

**Additional Context**

During the June, 2024 Hubverse retreat, we spent time exploring how we might standardize the creation of visualizations for Hubverse-compliant forecasting hubs (for this exercise, we used [these](https://respicast.ecdc.europa.eu/forecasts/?disease=1&target=1&season=1&location=AT&models=14&mode=detailed&display=date&scale=lin&ci=90&weeks-shown=4) [existing](https://viz.covid19forecasthub.org/) [visualizations](https://covid19scenariomodelinghub.org/viz.html) for [inspiration](http://flusightnetwork.io/)).

The exercise forced a useful discussion of trade-offs (_e.g._, aesthetics, ease of creation, re-use of existing code, ease of updates and maintenance, usability, performance). Some of this conversation is [captured in the retreat agenda](https://docs.google.com/document/d/1-hXi3RFH9zh46tT-Rzl1VVLjVmB0feVIaRJ9XvWy-90/).

Given that the time we invest in building a "dashboard" (a term we're using loosely to refer any of several items--see the poll options) should yield something useful to the community, the proposal is to pick a component to work on this summer and make an early version available to hubs that would be amenable to trying it out and informing the direction of future iterations.

**Note:** the plan is to target cloud-based hubs for this effort (_i.e._, data in AWS S3)

**Comment by elray1** [comment]
- I have voted for forecast/target visualization as what feels like the most appropriate place to get started on a dashboard that's oriented to the broadest audience of hub admins, modelers, and users of hub products (e.g. public health folks) for all hubs, if we are starting from scratch.
- in my mind, evaluation is the clear second step.  with viz+evaluation together, we'd have the core pieces of a dashboard targeted to those audiences, at least for forecast hubs.  I think evaluation is less of a priority than visualization because basically every hub will want a viz, but not every hub may want an evaluation dashboard.  and i'm less familiar with what it takes to build an evaluation for scenario projections.
- I recognize that this comment is not on the main theme of this poll, but -- I continue to feel like (a) taking the time to plan a bit more about the scope of this project before we jump in would be beneficial, and (b) ignoring existing solutions to eval and dashboarding more generally, in particular from the Delphi group and the European modeling hubs, puts us at risk of recreating existing work.

**Comment by bsweger** [comment]
We discussed this further in today's Hubverse developer meeting and clarified the goals of the effort:

1. create the first iteration of an "out of the box" dashboard with the goal of serving hub teams without the internal capacity or time to develop their own
2. create the first iteration in a modular way that can eventually be extended as needed

Because everyone seems to be in agreement that a visualization of predictions is a good starting point, am closing the poll.
